### ISSUE TITLE
Any plans for an Umbraco 9+ Version?

### ISSUE THREAD

**Issue by hfloyd** [vokseverk/Vokseverk.KeyValueEditor#5]
Hello! I really love using this package, but it seems to be abandoned? There is an open PR for an Umbraco 9 update (https://github.com/vokseverk/Vokseverk.KeyValueEditor/pull/4) that has been open for months.

Do you plan to continue to maintain this package going foreward? If not, would you mind if I set up a ported version myself?

Thanks!

**Comment by greystate** [comment]
Hi Heather - thanks for stoppin' by and for using the package 👋 

**Short answer:** Yes, I do plan to keep maintaining this package (I use it regularly myself - also in v10 sites).

**Slightly longer answer:** I just really haven't got around to understanding this whole nuget thing yet, and every time I've tried, I end up using significantly longer time fixing configs & googling errors etc. I literally have an item on my "What to learn over the Christmas holidays" list that says "[ ] Figure this nuget thing out for realz" 😬 — I have several packages that I'd like to make available, since they're "just" property editors that should be almost 1:1 compatible with v9+

The main reason for not having merged the PR though, is that doing so would completely destroy my own package build (I've had a similar PR for another package that does the exact same thing, so it's probably just what Visual Studio does when you pull in a project - nevertheless, it's a no-go for me). So I've left the PR open so people are able to grab that, until I figure this out.

Hope that explains things a bit

**Comment by hfloyd** [comment]
Thanks for replying 😊

That's great news!

I can't claim to have a perfect understanding/NuGet package setup... but I've got it working pretty well for my own packages. I keep my own setup pretty simple - using the [.csproj "Package" section](https://github.com/hfloyd/Dragonfly.Umbraco10.SchemaImporter/blob/3e1dea2223c2da9c89ad3f5db2a1ec2c05998018/src/Dragonfly/Dragonfly.csproj#L9), plus the build settings file ([example here...](https://github.com/hfloyd/Dragonfly.Umbraco10.SchemaImporter/blob/master/src/Dragonfly/build/Dragonfly.Umbraco10.SchemaImporter.targets)), and a ["Custom targets" file](https://github.com/hfloyd/Dragonfly.Umbraco10.SchemaImporter/blob/master/src/Custom.targets) to automatically push any "Release" builds to NuGet†

This is a template I use for the build settings file:
```
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
<!-- Find Replace: '_MY_PACKAGE_NAME_' with Actual custom package name-->
    <PropertyGroup>
        <_MY_PACKAGE_NAME_ContentFilesPath>$(MSBuildThisFileDirectory)..\content\App_Plugins\Dragonfly._MY_PACKAGE_NAME_\**\*.*</_MY_PACKAGE_NAME_ContentFilesPath>
    </PropertyGroup>

    <Target Name="Copy_MY_PACKAGE_NAME_Assets" BeforeTargets="Build">
        <ItemGroup>
            <_MY_PACKAGE_NAME_ContentFiles Include="$(_MY_PACKAGE_NAME_ContentFilesPath)" />
        </ItemGroup>
        <Message Text="Copying _MY_PACKAGE_NAME_ files: $(_MY_PACKAGE_NAME_ContentFilesPath) - #@(_MY_PACKAGE_NAME_ContentFiles->Count()) files"  Importance="high" />
        <Copy
            SourceFiles="@(_MY_PACKAGE_NAME_ContentFiles)"
            DestinationFiles="@(_MY_PACKAGE_NAME_ContentFiles->'$(MSBuildProjectDirectory)\App_Plugins\Dragonfly._MY_PACKAGE_NAME_\%(RecursiveDir)%(Filename)%(Extension)')"
            SkipUnchangedFiles="true" />

    </Target>

    <Target Name="Clear_MY_PACKAGE_NAME_Assets" BeforeTargets="Clean">
        <ItemGroup>
            <_MY_PACKAGE_NAME_Dir Include="$(MSBuildProjectDirectory)\App_Plugins\Dragonfly._MY_PACKAGE_NAME_\" />
        </ItemGroup>
        <Message Text="Clear old Dragonfly _MY_PACKAGE_NAME_ files"  Importance="high" />
        <RemoveDir Directories="@(_MY_PACKAGE_NAME_Dir)"  />
    </Target>

</Project>
```

It took me awhile to get everything figured out, but now it's generally running pretty smoothly for my packages. I'm sure better package developers out there would have critiques of this, but I prefer something I can work with easily to anything too complicated to maintain.

---
† You'll need to install your NuGet API key to your machine for that to "just work", here are some notes I have on that process:
[How to update the Key for VS usage](https://docs.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-setapikey)

TL;DR; Command:
`nuget setapikey MY_API_KEY`

**Comment by greystate** [comment]
Hey again,

So I've managed to get a build setup going (currently in a separate branch) where I can at least build a version for Umbraco 10. To start things off, I've thrown it on nuget.org as a "soft launch" to try it out in a couple of projects to see if I'm just breaking stuff 😆 ...

I've tried the [Umbraco create a testsite](https://docs.umbraco.com/v/10.x-lts/umbraco-cms/extending/packages/creating-a-package#installing-a-nuget-package) thing and made everything work in there, so hopefully that's enough. I can't do that for v9 as I can't test it as easily on my Mac because of the .net5 thing... but the build doesn't complain, so I'm thinking it'll work once I get to it.

Thanks again @hfloyd for pushing me over the fence with this - you made me want to give it a thorough try this past couple of weeks 👍 

It should be [available here](https://www.nuget.org/packages/Vokseverk.KeyValueEditor) on Nuget.org very soon and hopefully thus, in the Marketplace as well.

Oh and Happy New Year 🎉

**Comment by hfloyd** [comment]
Wohoo! Awesome news, @greystate 🍾
I'm working on a new v10 build this month, and can't wait to try it 😍 I will reply here is I see any issues. 

It is indeed a Happy New Year! 🤗

**Comment by greystate** [comment]
Thanks Heather — I'm particularly baffled about the UmbracoCms references as I'm very unsure if they will trigger an up- or downgrade of an existing install 🙈 ??

So yeah - just let me know 😬 😁

**Comment by greystate** [comment]
> this:
> <PackageReference Include="Umbraco.Cms.Core" Version="[10.0.1,11.0.0]" />
>
> Which basically means "must be >=10.0.1 and <11.0.0" And that seems to work well.

I did look into that at some point but was put off by the horridness of having to _start_ with an open square-bracket but _end_ with a closing-paren (😱) to do that (or I may have read the docs wrong...)

Anyways; thanks a lot for using this - I'll try to get some of my other packages on Nuget as well now that I seem to know how to do it :)

**Comment by greystate** [comment]
Thanks! Will have to look into that 🤔

**Comment by greystate** [comment]
Hi @hfloyd — totally forgot to mention that I put out v3.0.2 to hopefully fix the install errors 🤞.

**Comment by hfloyd** [comment]
Just installed 3.03 and it works! thanks!
